docs/sources/shared/wal-data-retention.md

```diff
@@ -0,0 +1,79 @@
+---
+aliases:
+- /docs/agent/shared/wal-data-retention/
+- /docs/grafana-cloud/agent/shared/wal-data-retention/
+- /docs/grafana-cloud/monitor-infrastructure/agent/shared/wal-data-retention/
+- /docs/grafana-cloud/monitor-infrastructure/integrations/agent/shared/wal-data-retention/
+canonical: https://grafana.com/docs/agent/latest/shared/wal-data-retention/
+description: Shared content, information about data retention in the WAL
+headless: true
+---
+
+The `prometheus.remote_write` component reuses the Prometheus Write-Ahead Log
```

Comment:
nit: I'd drop the reference that we're reusing the Prometheus WAL, since that's an implementation detail that could change (even though it probably won't).